src/utils/signer.ts

```diff
@@ -398,7 +427,27 @@ export function toSignableRequest(
     }
   });
 
-  // const reqBody = await req.body?.getReader().read();
+  // get the forwarded host
+  const forwardedHost = req.headers.get("x-forwarded-host");
+  if (globalConfig.trust_proxy && forwardedHost) {
+    const lastIp = req.headers.get("x-forwarded-for")?.split(",").pop()?.trim();
+    if (
+      !lastIp || !isIpInMultipleCidrRanges(lastIp, globalConfig.trusted_ips)
+    ) {
+      logger.critical(
```

Comment:
a warn may be better suited because critical usually means the error is so serious that it cannot be handled, which is not the case here